core/src/message/context.rs

```diff
@@ -328,6 +307,12 @@ impl MessageContext {
         delay: u32,
         reservation: Option<ReservationId>,
     ) -> Result<MessageId, Error> {
+        let new_outgoing_bytes = self
```

Comment:
to consider: move inside success branch. For example, this check is excess if Error::LateAccess scenario

Reply:
Agree. But I think it's more wide problem - because we have a bug: after payload `take()` in `send_commit` and `reply_commit` we do not set payload back in case an error occurs.

Reply:
done. issue created #3779